src/test/integration-tests/integration-tests.ts

```diff
@@ -0,0 +1,172 @@
+                baseUrl: "https://gaia.cruxpay.com/",
+                json: true,
+                method: "GET",
+                url: "1ATf5YwcEARWMCZdS8x3BXmkodkvnMW4Tf/cruxdev_client-config.json",
+            }
+            const mockGaiaRecord = [{"token":"eyJ0eXAiOiJKV1QiLCJhbGciOiJFUzI1NksifQ.eyJjbGFpbSI6eyJkNzhjMjZmOC03YzEzLTQ5MDktYmY2Mi01N2Q3NjIzZjhlZTgiOnsiYWRkcmVzc0hhc2giOiIxSFg0S3Z0UGRnOVFVWXdRRTFrTnFUQWptTmFERzd3ODJWIn19LCJpc3N1ZXIiOnsicHVibGljS2V5IjoiMDJlOGUxZGVhYmI3MGFlMGY4YzQzNjVkZDc0ZTE3ZjIwMmEwNzU3NjBmZjVhMDI4MDgzOGVmNGI2ZjUyMWM0NjE0In0sInN1YmplY3QiOnsicHVibGljS2V5IjoiMDJlOGUxZGVhYmI3MGFlMGY4YzQzNjVkZDc0ZTE3ZjIwMmEwNzU3NjBmZjVhMDI4MDgzOGVmNGI2ZjUyMWM0NjE0In19.fVony1HEW0oz5hewmBEfB0dIX5OqsYnrN5qqcnVLwOm3dHqTsCjyjbMcmvYQzlUuBdt2S8Lh8mK0Gdxm_5VFyw","decodedToken":{"header":{"typ":"JWT","alg":"ES256K"},"payload":{"claim":{"d78c26f8-7c13-4909-bf62-57d7623f8ee8":{"addressHash":"MALICIOUS_BTC_ADDRESS"}},"issuer":{"publicKey":"02e8e1deabb70ae0f8c4365dd74e17f202a075760ff5a0280838ef4b6f521c4614"},"subject":{"publicKey":"02e8e1deabb70ae0f8c4365dd74e17f202a075760ff5a0280838ef4b6f521c4614"}},"signature":"fVony1HEW0oz5hewmBEfB0dIX5OqsYnrN5qqcnVLwOm3dHqTsCjyjbMcmvYQzlUuBdt2S8Lh8mK0Gdxm_5VFyw"}}]
+            const mockHttpJSONRequest = sandbox.stub(utils, "httpJSONRequest").callThrough().withArgs(sinon.match(requestOptions)).resolves(mockGaiaRecord);
+            // calling the method
+            let raisedError;
+            try {
+                const resolvedAddress = await cruxWalletClient.resolveCurrencyAddressForCruxID(testCruxId.toString(), "btc");
+            } catch (error) {
+                raisedError = error;
+            }
+            // expectations
+            expect(mockHttpJSONRequest.calledOnce).to.be.true;
+            expect(raisedError).to.be.instanceOf(CruxClientError);
+            expect(raisedError["errorCode"]).to.be.equal(PackageErrorCode.GaiaRecordIntegrityFailed);
+        })
+        it("resolving address of a user whose gaiaHub is tampered (malicious gaiaHub info response)", async () => {
+            const requestOptions = {
+                baseUrl: "https://hub.cruxpay.com",
+                url: "/hub_info",
+            };
+            const mockGaiaHubInfoResponse = {"challenge_text":"[\"gaiahub\",\"0\",\"hub.cruxpay.com\",\"blockstack_storage_please_sign\"]","latest_auth_version":"v1","max_file_upload_size_megabytes":20,"read_url_prefix":"https://gaia.cruxpay.com?fakeKey=${console.log('test')}"};
+            const mockHttpJSONRequest = sandbox.stub(utils, "httpJSONRequest").callThrough().withArgs(sinon.match(requestOptions)).resolves(mockGaiaHubInfoResponse);
+            const resolvedAddress = await cruxWalletClient.resolveCurrencyAddressForCruxID(testCruxId.toString(), "btc");
+            // @ts-ignore
+            expect(mockHttpJSONRequest.parent.neverCalledWith(sinon.match({ baseUrl: "https://gaia.cruxpay.com?fakeKey=${console.log('test')}" }))).to.be.true;
```

Comment:
expect(mockHttpJSONRequest.parent.CalledWith(sinon.match({ baseUrl: "https://gaia.cruxpay.com}))).to.be.true;

Will the above be true for the test case? If yes in addition to negative test case can we add this/equivalent of the same as well?